eth/api.go

```diff
@@ -664,6 +664,7 @@ func (s *PublicBlockChainAPI) doCall(args CallArgs, blockNr rpc.BlockNumber) (st
 	if msg.gasPrice.Cmp(common.Big0) == 0 {
 		msg.gasPrice = new(big.Int).Mul(big.NewInt(50), common.Shannon)
```

Comment:
Since this is a call, can't we set the gas price to 0? Seems that 50 shannon is very arbitrary and would just cause strange troubles later on. Similarly a few lines above the gas is set to 5M if it's not specified, maybe we could figure out something better than a hard coded value?